File: Sorting/InsertionSort.java
// Java Code to illustate Insertion sorting

import java.util.*;
import java.io.*;

public class InsertionSort{

public static void main(String args[]){
  //Declarartion of required variables
  int size;
  Scanner sc=new Scanner(System.in);
  //Reading Input
  System.out.println("Input :\n");
  System.out.println("Enter the size of an array : \n");
   size=sc.nextInt();
   sc.nextLine();
    int arr[]=new int[size];
  System.out.println("Enter the elements of an array :\n");
  for(int index = 0; index < size; index++) {
          arr[index]=sc.nextInt();
  }
  for (int index =1; index < size; index++) {
    int temp=arr[index];
    int j=index-1;
    while(j>=0 && arr[j] > temp) {
        //Swapping the elements, temp is used to store the temporary variable
        arr[j + 1] = arr[j];
        j=j-1;
      }
      arr[j+1]=temp;
    }
  //Displaying Output
  System.out.println("Output :\n");
  System.out.println("The sorted array is :\n");
  for (int index = 0; index < size; index++) {
   System.out.print(arr[index]+" ");
  }
 
}
}

/*
 Input:
 Enter the size of an array: 4
 Enter the elements of an array :
 10 8 1 4
 Output :
 1 4 8 10 
 Time Complexity :O(n^2)
 Space Complexity :O(1)
*/

File: Sorting/Insertion Sort.cpp
#include<iostream>
using namespace std;
 
void insertionSort(int arr[]) {
  int key;
  int j = 0;
  for (int i = 1; i<5; i++) {
    key = arr[i];
    j = i - 1;
    while (j >= 0 && arr[j] > key) {
      arr[j + 1] = arr[j];
      j = j - 1;
    }
    arr[j + 1] = key;
  }
 
}
 
int main() {
  int myarray[5];
  cout<<"Enter 5 integers in any order"<<endl;
  for (int i = 0; i<5; i++) {
    cin>>myarray[i];
  }
 
  cout<<"Before Sorting: "<<endl;
  for (int i = 0; i<5; i++) {
    cout<<myarray[i]<<" ";
  }
 
  insertionSort(myarray);
 
  cout<<endl<<"After Sorting: "<<endl;
  for (int i = 0; i<5; i++) {
    cout<<myarray[i]<<" ";
  }
 
  return 0;
}
